app/domain/authentication/authn_jwt/vendor_configurations/configuration_interface.rb

```diff
@@ -0,0 +1,16 @@
+module Authentication
+  module AuthnJwt
+    module VendorConfigurations
+      # Interface containing JWT configuration functions to be implemented in JWTConfiguration class for a vendor
+      class ConfigurationInterface
+        def initialize(authenticator_input); end
+
+        def jwt_identity; end
+
+        def validate_restrictions; end
+
+        def validate_and_decode_token; end
```

Comment:
Since ruby is a duck-typed language, we should never create "interface" classes like this in ruby.  This is weird if you're coming from a background with static language like Java or C#, but an interface in ruby is implicit.  You can document it, but you don't create an empty class for it.  You can also point to a concrete implementation and say something like, "this method expects an object like X".

Reply:
Thanks. Removed interfaces